lib/full-site-editing/template-parts.php

```diff
@@ -201,3 +201,36 @@ function gutenberg_filter_template_part_area( $type ) {
 	trigger_error( $warning_message, E_USER_NOTICE );
 	return WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
 }
+
+/**
+ * Print a template-part.
+ *
+ * @param string $part The template-part to print. Use "header" or "footer".
+ *
+ * @return void
+ */
+function gutenberg_block_template_part( $part ) {
+	$template_part = gutenberg_get_block_template( get_stylesheet() . '//' . $part, 'wp_template_part' );
+	if ( ! $template_part || empty( $template_part->content ) ) {
+		return;
+	}
+	echo do_blocks( $template_part->content );
```

Comment:
Right now, I think we are just rendering the internal block content of the template part.  But since we aren't rendering a template part block, none of the template part specific logic is being utilized and there is no wrapping being applied.

If we do something like what @gziolo suggests and instead render a template part block with attributes (slug and theme), it should work as expected.  I think the problem with this suggestion is that for non-block-based themes this block is never registered anyways? 🤔 

I think this current approach is fine if we are ok with the fact that Classic themes will need to write their template-part markup slightly different than FSE themes (i.e. include a wrapper element to set the appropriate semantic html tag, etc.).  However, if we want Classic and FSE themes to write their template part markup the same way and have classic utilize the 'area' terms set up in themes.json, we should probably find a way to go with what @gziolo is suggesting?

Reply:
Classic themes already add these tags... 
Typically a theme will have something like this in their `header.php` file:
```php
<header id="masthead" class="site-header" role="banner">
	<div class="site-branding">
		LOGO HERE
	</div>
	
	<?php if ( has_nav_menu( 'primary' ) ) : ?>
		<nav id="site-navigation" class="primary-navigation" role="navigation">
			MENU HERE
		</nav>
	<?php endif; ?>
</header>
```
With the functions added here, they can move all contents of their header inside a template-part and convert their code to something like this instead:
```php
<header id="masthead" class="site-header" role="banner">
	<?php gutenberg_block_header_area(); ?>
</header>
```
The template-part itself is identical to what it would be in an block theme.

Is the problem that they could also delete the wrappers from their code, and we could instead include them in the `gutenberg_block_header_area()` call?
We can definitely do that if you believe it will help authors! This is just a tool to help classic themes start migrating parts of their themes to a block structure, so anything that helps them do that is on the table. Not adding the wrappers simply seemed like the safer way to do it at the time.

Reply:
Thanks for clarifying!  It seems like its set up pretty well then.

> Is the problem that they could also delete the wrappers from their code, and we could instead include them in the gutenberg_block_header_area() call?

I don't think there is a problem specifically then.  I was thinking the idea that was recommended above `do_blocks( sprintf( '<!-- wp:template-part %s /-->', json_encode( $attributes ) ) );` might be easier, but to get the right wrapper with that you would need to define that template part's area in themes.json as well.  So its probably more of a hassle than its worth considering the way things currently work?

> We can definitely do that if you believe it will help authors!

Il leave that decision up to you.  It seems like this is a good way to go for now. 😁 

Thanks again!